[Bombbird2001] iP

Bombbird2001 opened the pull request:
## SHIBA-BOT
> Your personal Shib-uddy

SHIBA-BOT helps organise your day, with
* A text-based UI
* An almost non-existent learning curve
* Cuteness

### Get started now, by
1. Downloading it [here](https://github.com/Bombbird2001/ip/releases/tag/v1.1)
2. Double click on the file (or run ```java -jar SHIBA-BOT.jar```)
3. Add and manage your tasks!

Best part, it's **free**!

### Features
- [x] Adding, deleting tasks
- [x] Marking, unmarking tasks
- [x] 3 different task types
- [x] Play with SHIBA-BOT using ```pat```, ```bellyrub``` and ```boop```!
- [ ] More commands to play with SHIBA-BOT!
---
And here's the code in the the main method, just because:
```java
public static void main(String[] args) {
        new Shiba("SHIBA-BOT", "./shibaData/tasks.txt");

        Application.launch(MainWindow.class);

        MainWindow.getInstance().cleanUp();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Bombbird2001 committed:
Level-0 changes

Bombbird2001 committed:
Level-1 changes

Bombbird2001 committed:
Level-2 changes

Bombbird2001 committed:
Level-3 changes

Bombbird2001 committed:
Level-4 changes

Bombbird2001 committed:
TextUiTesting changes

Bombbird2001 committed:
Level-5 changes

Bombbird2001 committed:
Level-6 changes

Bombbird2001 committed:
Enum changes

Bombbird2001 committed:
Add saving & loading of tasks to/from disk file

Bombbird2001 committed:
Merge branch 'branch-Level-7'

Bombbird2001 committed:
Add datetime or date formats for deadlines, events

Bombbird2001 committed:
Update tests to account for datetime input

Bombbird2001 committed:
Merge branch 'branch-Level-8'

Bombbird2001 committed:
Extract classes to parser, UI, command, storage categories

Bombbird2001 committed:
Merge branch 'A-MoreOOP'

Bombbird2001 committed:
Move all classes to packages

Bombbird2001 committed:
Merge branch 'branch-A-Packages'

Bombbird2001 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Bombbird2001 committed:
Add gradle, with junit enabled

Bombbird2001 committed:
Add tests for DateOptionalTime class

Bombbird2001 committed:
Add tests for TodoCommand class

Bombbird2001 committed:
Update jar name

Bombbird2001 committed:
Update some method JavaDocs

Bombbird2001 committed:
Add line breaks for long lines

Bombbird2001 committed:
Add support for find command

Bombbird2001 committed:
Add automated test for find command

Bombbird2001 committed:
Merge branch 'branch-A-JavaDoc'

Bombbird2001 committed:
Merge branch 'branch-A-CodingStandard'

Bombbird2001 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/shiba/tasks/PersistentTaskList.java

Bombbird2001 committed:
Fixed checkstyle violations

Bombbird2001 committed:
Add checkstyle version; checkstyle violations have been fixed earlier

Bombbird2001 committed:
Add initial layout for UI

Bombbird2001 committed:
Implement full UI including user input, bot response

Bombbird2001 committed:
Fix checkstyle violations, prevent user from sending empty input

Bombbird2001 committed:
Add user, bot images

Bombbird2001 committed:
Print start message only after window startup

Bombbird2001 committed:
Remove unnecessary thread

Bombbird2001 committed:
Add pat, bellyrub commands

Bombbird2001 committed:
Change wildcard import for coding standard

Bombbird2001 committed:
Add better indentation for different text; add press enter to send command

Bombbird2001 committed:
Add missing inheritdoc to overriding methods

Bombbird2001 committed:
Fix program not ending when closed using close button

Bombbird2001 committed:
Update README.md; varargs have already been added in an earlier commit

Bombbird2001 committed:
Change gradle to build with Java 11

zhengyup commented on src/main/java/shiba/commands/BellyRubCommand.java:
```diff
@@ -0,0 +1,33 @@
+public class BellyRubCommand extends ShibaCommand {
+    private static final Random random = new Random();
+    private static final String msg1 = "Woof! More please!";
+    private static final String msg2 = "<Happy Shiba noises>";
```
Could these strings be named with more informative names?

zhengyup commented on src/main/java/shiba/Shiba.java:
```diff
@@ -0,0 +1,70 @@
+     * @param input The user input.
+     */
+    public void processUserInput(String input) {
+        boolean quit = parser.processUserInput(input);
```
Could this variable be named with a more boolean name ?

zhengyup commented on src/main/java/shiba/commands/ShibaCommand.java:
```diff
@@ -0,0 +1,88 @@
+    /**
+     * The types of commands that Shiba can execute.
+     */
+    public enum CommandType {
```
I like this abstraction and clear names given! especially belly rub

zhengyup commented on src/main/java/shiba/commands/ShibaCommand.java:
```diff
@@ -0,0 +1,88 @@
+        if (cmd.length < 2) {
+            throw new InvalidCommandException("Please specify a task number!");
+        }
+
```
could this blank line be omitted?

zhengyup commented on src/main/java/shiba/datetimeformats/DateOptionalTime.java:
```diff
@@ -0,0 +1,75 @@
+        if (dateTime != null) {
+            return dateTime.toString();
+        }
+
```
could this blank line be omitted ?

zhengyup commented on src/main/java/shiba/parsers/CommandParser.java:
```diff
@@ -0,0 +1,88 @@
+                    input.split(" ")[0].toUpperCase());
+            ShibaCommand shibaCommand;
+            switch (command) {
+            case LIST:
```
I like how you formatted the switch statement correctly here!

zhengyup reviewed (commented):
LGTM! Just a few nits to fix. Shiba bot sounds very cute

Bombbird2001 committed:
Add assertions

We are not able to pass an instance of Shiba to MainWindow, and vice
versa, so we have singletons for both of these classes instead. Hence,
we check that the singleton instance does not already exist when
creating a new instance.

We also assert that a file exists after creating it with no errors.

Bombbird2001 committed:
Improve code style, add indent for tasks in find command

Currently, tasks are not indented in the find command.
Indented block for tasks will make them easier to read.

Bombbird2001 committed:
Use streams for find command

Improved readability and more concise code using streams
as compared to using a conventional for loop

Bombbird2001 committed:
Merge pull request #1 from Bombbird2001/branch-A-Assertions

Add assertions

Bombbird2001 committed:
Merge branch 'master' into branch-A-CodeQuality

Bombbird2001 committed:
Merge pull request #2 from Bombbird2001/branch-A-CodeQuality

Improve code style, add indent for tasks in find command

Bombbird2001 committed:
Merge branch 'master' into branch-A-Streams

Bombbird2001 committed:
Merge pull request #3 from Bombbird2001/branch-A-Streams

Use streams for find command

Bombbird2001 committed:
Add CI (might not work, still testing)

Bombbird2001 committed:
Add C-Help

Help messages help unfamiliar users correctly execute the commands.

Bombbird2001 committed:
Merge pull request #4 from Bombbird2001/branch-C-Help

Add C-Help

Bombbird2001 committed:
Correct branch name from main to master

Bombbird2001 committed:
Correct some constant variable names

Bombbird2001 committed:
Merge pull request #5 from Bombbird2001/branch-CI-Test

Correct some constant variable names

dlathyun commented on src/main/java/shiba/commands/BellyRubCommand.java:
```diff
@@ -0,0 +1,33 @@
+ */
+public class BellyRubCommand extends ShibaCommand {
+    private static final Random RANDOM = new Random();
+    private static final String REPLY_MSG_1 = "Woof! More please!";
+    private static final String REPLY_MSG_2 = "<Happy Shiba noises>";
```
I like it that you used common prefix for these Strings!

dlathyun commented on src/main/java/shiba/commands/HelpCommand.java:
```diff
@@ -0,0 +1,128 @@
+/**
+ * Represents a command to display help messages.
+ */
+public class HelpCommand extends ShibaCommand {
```
I think that it is nice that there is a "help" command that helps the user better understand the commands!

dlathyun reviewed (commented):
Overall good code quality :) Has unique and cute commands, and overall the code is well designed. I liked how the files were extracted into different packages/classes (even the parser and datetime classes).

Bombbird2001 committed:
Make UI window resizable, align user text to right

Bombbird2001 committed:
Improve UI including colours, speech bubbles and spacing

Bombbird2001 committed:
Code quality fixes

Bombbird2001 committed:
Merge branch 'master' into branch-A-BetterGui

Bombbird2001 committed:
Add new Boop command

Shiba-like personality has been added earlier to SHIBA-BOT,
so here's another random command to add on to it.

Bombbird2001 committed:
Add check for event, end datetime must >= start datetime

Bombbird2001 committed:
Add error message if parsing error occurs when loading from save file,
reduce font size

Bombbird2001 committed:
Fix file load errors not being printed in error message style

Bombbird2001 committed:
Improve unknown command error message to include the offending command,
fix file reader, writer to work with UTF-8 charset

Bombbird2001 committed:
Add user guide, image

Bombbird2001 committed:
Merge branch 'branch-A-MoreErrorHandling'

Bombbird2001 committed:
Strip spaces when parsing names and datetimes, add more tests

Bombbird2001 committed:
Add more command tests

Bombbird2001 committed:
Add parser tests

Bombbird2001 committed:
Fix quantum renderer crash on Linux
